src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/locationfinder/AppointmentLocationsService.kt

```diff
@@ -32,7 +32,15 @@ class AppointmentLocationsService(
     else
       videoLinkBookingRepository
         .findAllById(vlbIdsToExclude)
-        .flatMap { booking -> booking.toAppointments().map { it.appointmentId } }
+        .asSequence()
+        .flatMap {
+          sequenceOf(
+            it.main.appointmentId,
+            it.post?.appointmentId,
+            it.pre?.appointmentId
+          )
+            .filterNotNull()
```

Comment:
meh.. could add a `toAppointmentIds`method to VLB, like the current `toAppointmentDtos`

that uses `listOfNotNull` presumably there is a `setOfNotNull`  too.